--- a/apps/jscad-web/examples/openscad/dotscad/experimental/tile_truchet.scad
+++ b/apps/jscad-web/examples/openscad/dotscad/experimental/tile_truchet.scad
@@ -17,17 +17,10 @@
 				[for(x = x_range) my[x]]
 		]
 	)
-	is_undef(seed) ? [
+	[
         for(y = y_range)
-        let(my = m[y])      
-            for(x = x_range)  
+        let(my = m[y])
+            for(x = x_range)
             if(my[x] == 1)
-                [x, y, choose(nums)]
-        
-    ] : [
-        for(y = y_range)  
-        let(my = m[y])      
-            for(x = x_range) 
-            if(my[x] == 1)
-            [x, y, choose(nums, x + y * rows + seed)]
+            [x, y, choose(nums, x + y * rows + (is_undef(seed) ? 0 : seed))]
     ];
\ No newline at end of file
