--- a/apps/jscad-web/examples/openscad/dotscad/examples/circle_packing/packing_circles.scad	2026-03-31 10:23:54.000000000 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/circle_packing/packing_circles.scad	2026-04-06 13:30:00.000000000 -0400
@@ -7,8 +7,8 @@ point_numbers = 100;
     
 points = [
     for(i = [0:point_numbers - 1])
-    [rand(0, size.x), rand(0, size.y)]
-]; 
+    [rand(0, size.x, i * 2), rand(0, size.y, i * 2 + 1)]
+];
 
 circles = circle_packing(points, min_radius);
 mr = max([for(c = circles) c[1]]);
--- a/apps/jscad-web/examples/openscad/dotscad/examples/worley_noise_ball.scad	2026-03-31 10:23:54.037609865 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/worley_noise_ball.scad	2026-04-06 09:44:49.642720716 -0400
@@ -7,7 +7,7 @@
 dist = "euclidean"; // [euclidean, manhattan, chebyshev, "border"] 
 noise_style = "CELL_R";    // [CELL_R, NOISE]
 noise_factor = 1;
-seed = rands(0, 255, 1)[0];
+seed = rands(0, 255, 1, 42)[0];
 
 module worley_noise_ball() {
     theta_tau_lt = [ 
--- a/apps/jscad-web/examples/openscad/dotscad/examples/hollow_out/hollow_out_cylinder.scad	2026-03-31 10:23:54.020609632 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/hollow_out/hollow_out_cylinder.scad	2026-04-06 09:44:53.271771304 -0400
@@ -9,8 +9,8 @@
 radius = 15;
 fn = 12;
 
-xs = rands(0, size[0], pt_nums);
-ys = rands(0, size[1], pt_nums);
+xs = rands(0, size[0], pt_nums, 42);
+ys = rands(0, size[1], pt_nums, 43);
 half_fn = fn / 2;
 dx = size[0] / fn;
 dy = size[1] / half_fn;
--- a/apps/jscad-web/examples/openscad/dotscad/examples/voronoi/voronoi_bracelet.scad	2026-03-31 10:23:54.037199842 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/voronoi/voronoi_bracelet.scad	2026-04-06 09:44:56.359814350 -0400
@@ -16,8 +16,8 @@
     x = 2 * PI * r - thickness;
     y = height;
 
-    xs = rands(0, x, n);
-    ys = rands(0, y, n);
+    xs = rands(0, x, n, 42);
+    ys = rands(0, y, n, 43);
 
     points = [for(i = [0: len(xs) - 1]) [xs[i], ys[i]]];
         
--- a/apps/jscad-web/examples/openscad/dotscad/examples/voronoi/worley_vase.scad	2026-03-31 10:23:54.037609865 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/voronoi/worley_vase.scad	2026-04-06 09:44:59.414856936 -0400
@@ -22,7 +22,7 @@
 worley_vase(beginning_radius, height, thickness, fn, amplitude, curve_step, sample_scale, grid_w, dist, bottom, epsilon);
 
 module worley_vase(beginning_radius, height, thickness, fn, amplitude,curve_step, sample_scale, grid_w, dist, bottom, epsilon) {
-    seed = rand() * 1000;
+    seed = rand(0, 1, 42) * 1000;
 	section = shape_circle(radius = beginning_radius, $fn = fn);
 	pt = [beginning_radius, 0, 0];
     h_s = height / 140;
--- a/apps/jscad-web/examples/openscad/dotscad/examples/tiles/magic_apartment.scad	2026-03-31 10:23:54.034987367 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/tiles/magic_apartment.scad	2026-04-06 09:45:03.000906924 -0400
@@ -19,7 +19,7 @@
 			for(y = [0:rows])
 			[
 				for(x = [0:columns]) 
-				let(rs = rands(0, 1, 3))
+				let(rs = rands(0, 1, 3, z * 10000 + y * 100 + x))
 				[round(rs[0]), round(rs[1]), round(rs[2])]
 			]
 		]
--- a/apps/jscad-web/examples/openscad/dotscad/examples/hypnotic_squares.scad	2026-03-31 10:23:54.021728589 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/hypnotic_squares.scad	2026-04-06 09:45:11.449024687 -0400
@@ -10,7 +10,7 @@
 fn = 12;
 
 module hypnotic_squares(x_grids, y_grids, grid_size, final_size, line_width) {
-    function random() = rands(0, 1, 1)[0];
+    function random() = rands(0, 1, 1, 42)[0];
 
     dirs = [-1, 0, 1];
     max_steps = ceil((grid_size - final_size) / (line_width * 2));
@@ -42,8 +42,8 @@
         }
     }
 
-    rand_lt = [for(x = [0:x_grids - 1]) 
-                  [for(y = [0:y_grids - 1]) rands(0, 1, 2)]
+    rand_lt = [for(x = [0:x_grids - 1])
+                  [for(y = [0:y_grids - 1]) rands(0, 1, 2, x * y_grids + y)]
               ];
               
     module grids() {
--- a/apps/jscad-web/examples/openscad/dotscad/examples/taiwan/voronoi_taiwan.scad	2026-03-31 10:23:54.033609810 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/taiwan/voronoi_taiwan.scad	2026-04-06 09:53:45.327193949 -0400
@@ -8,8 +8,8 @@
     taiwan = shape_taiwan(120, distance = 1);

     n = 40;
-    xs = rands(-15, 15, n);
-    ys = rands(-30, 40, n);
+    xs = rands(-15, 15, n, 42);
+    ys = rands(-30, 40, n, 43);
     pts = [
         for(i = [0:n - 1])
             let(p = [xs[i], ys[i]])
@@ -21,7 +21,7 @@

     cells = tri_delaunay(all, ret = "VORONOI_CELLS");
     for(i = [0:len(cells) - 1]) {
-        color(rands(0, 1, 3))
+        color(rands(0, 1, 3, i))
         translate(all[i])
         linear_extrude(2, scale = 0.8)
         translate(-all[i])
--- a/apps/jscad-web/examples/openscad/dotscad/examples/maze/devil_maze.scad	2026-03-31 10:23:54.000000000 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/maze/devil_maze.scad	2026-04-06 09:26:50.963722686 -0400
@@ -75,7 +75,7 @@ module devil_maze() {
     ];

     cells = vrn2_cells_space([r, r] * 2 + [1, 1], 1);
-    seed = rands(0, 1000, 1)[0];
+    seed = rands(0, 1000, 1, 42)[0];
     for(cell = cells) {
         cell_pt = cell[0];
         cell_poly = cell[1];
--- a/apps/jscad-web/examples/openscad/dotscad/examples/maze/giant_maze.scad	2026-03-31 10:23:54.000000000 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/maze/giant_maze.scad	2026-04-06 09:26:50.963722686 -0400
@@ -54,7 +54,7 @@ module giant_maze() {
     ];

     cells = vrn2_cells_space([len(m[0]), len(m)], 1);
-    seed = rands(0, 1000, 1)[0];
+    seed = rands(0, 1000, 1, 42)[0];
     for(cell = cells) {
         cell_pt = cell[0];
         cell_poly = cell[1];
--- a/apps/jscad-web/examples/openscad/dotscad/examples/maze/maze_city.scad	2026-03-31 10:23:54.000000000 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/maze/maze_city.scad	2026-04-06 09:26:50.963722686 -0400
@@ -13,7 +13,7 @@ maze_city(rows, columns, skyscraper_prs);
 module maze_city(rows, columns, skyscraper_prs) {
     mask = [for(r = [0:rows - 1])
 	    [for(c = [0:columns - 1])
-		rand() < skyscraper_prs ? 0 : 1]
+		rand(0, 1, r * columns + c) < skyscraper_prs ? 0 : 1]
 	];

     cells = mz_square(rows, columns, [0, 0], init_cells = mz_square_initialize(rows, columns, mask));
--- a/apps/jscad-web/examples/openscad/dotscad/examples/voronoi/voronoi_vase.scad	2026-03-31 10:23:54.000000000 -0400
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/voronoi/voronoi_vase.scad	2026-04-06 09:26:50.963722686 -0400
@@ -33,7 +33,7 @@ module voronoi_vase(r, h, thickness, num_of_pts, fn, profile_step) {
     ];

     pts =  [for(sect = sections) each sect];
-    indices = rands(0, len(pts) - 1, num_of_pts - 4);
+    indices = rands(0, len(pts) - 1, num_of_pts - 4, 42);

     last_section_i = len(sections) - 1;
--- a/apps/jscad-web/examples/openscad/dotscad/examples/distorted_vase.scad
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/distorted_vase.scad
@@ -21,7 +21,7 @@ epsilon = 0.000001;
 distorted_vase(beginning_radius, thickness, fn, amplitude, curve_step, smoothness, perlin, epsilon);
 
 module distorted_vase(beginning_radius, thickness, fn, amplitude,curve_step, smoothness, perlin, epsilon) {
-	seed = rand() * 1000;
+	seed = 42;
 	section = shape_circle(radius = beginning_radius, $fn = fn);
 	pt = [beginning_radius, 0, 0];
 
--- a/apps/jscad-web/examples/openscad/dotscad/examples/perlin_noise_cylinder.scad
+++ b/apps/jscad-web/examples/openscad/dotscad/examples/perlin_noise_cylinder.scad
@@ -20,7 +20,7 @@ module perlin_noise_cylinder(radius, height, thickness_scale, step) {
             ]
     ];
 
-    seed = rand(0, 256);
+    seed = rand(0, 256, 42);
     leng_row = len(surface_inside[0]);
     surface_outside = [
             for(ri = [0:len(surface_inside) - 1])
