--- a/apps/jscad-web/examples/openscad/dotscad/voronoi/vrn2_cells_space.scad
+++ b/apps/jscad-web/examples/openscad/dotscad/voronoi/vrn2_cells_space.scad
@@ -14,7 +14,7 @@
 
 function vrn2_cells_space(size, grid_w, seed) = 
     let(
-        sd = is_undef(seed) ? rands(0, 255, 1)[0] : seed,
+        sd = is_undef(seed) ? 0 : seed,
         region_size = grid_w * 3,
         half_region_size = region_size * 0.5,
         shape = shape_square(grid_w * 3),
